### PR TITLE
CLDC-3249 Clear invalidated earnings values

### PR DESCRIPTION
**Pull request by kosiakkatrina**

### CODE REVIEW

File: spec/models/validations/financial_validations_spec.rb [hunk 1]
```diff
@@ -293,6 +300,20 @@
           expect(record.errors["ecstat#{n}"]).to be_empty
         end
       end
+
+      context "when the net income is lower than the hard min for their employment status" do
+        it "adds an error" do
+          record.startdate = Time.zone.local(2022, 5, 1)
```
Comment (natdeanlewissoftwire) [inline spec/models/validations/financial_validations_spec.rb:306]: I think it's worth mentioning in the context/it description this is a 22/23 log given the rest in the file are 23/24

File: spec/lib/tasks/clear_invalidated_earnings_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    before do
+      Rake.application.rake_require("tasks/clear_invalidated_earnings")
+      Rake::Task.define_task(:environment)
+      task.reenable
+      FormHandler.instance.use_real_forms!
+    end
+
+    context "when the rake task is run" do
+      context "and there are 2023 logs with invalid earnings" do
```
Comment (natdeanlewissoftwire) [inline spec/lib/tasks/clear_invalidated_earnings_spec.rb:16]: it definitely will pass, but I wouldn't hate there being a test ensuring 2022 logs/valid 2023 logs remain unchanged and valid.

**Review by natdeanlewissoftwire — APPROVED** [review]

File: spec/models/validations/financial_validations_spec.rb (hunk 1 above)
Comment (kosiakkatrina) [reply spec/models/validations/financial_validations_spec.rb:306]: Ah, yes, forgot to rename it, thanks!

File: spec/lib/tasks/clear_invalidated_earnings_spec.rb (hunk 2 above)
Comment (kosiakkatrina) [reply spec/lib/tasks/clear_invalidated_earnings_spec.rb:16]: Yep! Will add that!

**Pull request merged**